testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/MyResourcesPage.java

```diff
@@ -34,25 +55,99 @@ public void clickShareButton(int row) {
+    }
+
+    private void waitForModalFadeIn() {
+        waitGui().until().element(By.className("pf-c-modal-box")).is().present();
+    }
+
+    public void clickRemoveButton(int row) {
+        final WebElement webElement = driver.findElement(By.id("ex-expand" + row));
+
+        //first button share 2rd is the edit button and 3rd is remove
+        webElement.findElements(By.tagName("button")).get(2).click();
+    }
+
+    public void clickPendingRequest(String resourceName) {
+        final WebElement pendingRequestButton = driver.findElement(By.id("shareRequest-" + resourceNameToId(resourceName)));
+        assertEquals("1", pendingRequestButton.getText());
+        pendingRequestButton.click();
+    }
+
+    public void acceptRequest(String resourceName, int row) {
+        clickApproveDenyButton(resourceName, row, true);
+    }
+
+    public void denyRequest(String resourceName, int row) {
+        clickApproveDenyButton(resourceName, row, false);
+    }
+
+    private void clickApproveDenyButton(String resourceName, int row, boolean approve) {
+        final By id = By.id(String.format("%s-%d-shareRequest-%s", approve ? "accept" : "deny", row, resourceNameToId(resourceName)));
+        driver.findElement(id).click();
+        waitForModalFadeOut();
+    }
+
+    private String resourceNameToId(String resourceName) {
+        return resourceName.replace(" ", "-");
+    }
+
+    public void clickSignOut() {
+        driver.findElement(By.id("signOutButton")).click();
+    }
+
+    public void clickNextPage() {
+        final WebElement webElement = driver.findElements(By.className("pf-m-primary")).get(1);
+        assertEquals("Next>", webElement.getText());
+        webElement.click();
+    }
+
+    public void clickSharedWithMeTab() {
+        final WebElement sharedWithMe = driver.findElement(By.id("pf-tab-1-sharedwithMe"));
+        sharedWithMe.click();
+
+        final WebElement tab = sharedWithMe.findElement(By.xpath("./.."));
+        //test to see that the tab is really clicked
+        assertEquals("pf-c-tabs__item pf-m-current", tab.getAttribute("class"));
+    }
+
+    public boolean containsResource(String resourceName) {
+        return driver.findElement(By.id("sharedResourcesList")).getText().contains(resourceName);
     }
 
     public void createShare(String userName) {
         driver.findElement(By.id("username")).sendKeys(userName);
         driver.findElement(By.id("add")).click();
         driver.findElement(By.id("remove_pf-random-id-1")).click();
         driver.findElement(By.id("done")).click();
+        waitForModalFadeOut();
     }
 
     public void removeAllPermissions() {
         driver.findElement(By.id("remove_pf-random-id-1")).click();
         driver.findElement(By.id("remove_pf-random-id-2")).click();
         driver.findElement(By.id("done")).click();
+        waitForModalFadeOut();
+    }
+
+    public ContentAlert alert() {
+        return alert;
+    }
```

Comment:
Why to override this method? It seems identical (the `alert` field as well) to the parent.

Reply:
oops